### PR TITLE
fix: log warnings for full payload

### PR DESCRIPTION
**Pull request by majorgreys**
This fixes the lack of a log warning for sending a single trace to the agent that is larger than the max payload size. At the moment there is a log warning when a trace is larger than the max payload size is sent but only when alongside other traces.

### CODE REVIEW

File: tests/test_integration.py [hunk 1]
```diff
@@ -294,6 +297,60 @@ def test_send_many_traces(self):
 
         self._send_traces_and_check(traces, 2)
 
+    def test_send_single_trace_max_payload(self):
+        payload = Payload()
+
+        # compute number of spans to create to surpass max payload
+        trace = [Span(self.tracer, "child.span")]
+        trace_size = len(payload.encoder.encode_trace(trace))
+        num_spans = int(math.floor(payload.max_payload_size / trace_size))
```
Comment (brettlangdon) [inline tests/test_integration.py:306]: I wonder if there is a better way to test this, like set the payload max size to like 1b or something super small?

**Comment by majorgreys** [comment]
> we don't try to send an empty payload, correct?

Only send encoded traces if payload is not empty 

https://github.com/DataDog/dd-trace-py/pull/1543/files#diff-268b80cdf25949b4c50462f171fe5c5dL233-L234

File: tests/test_integration.py (hunk 1 above)
Comment (majorgreys) [reply tests/test_integration.py:306]: This solution was copied over from `test_payload.py`:

https://github.com/DataDog/dd-trace-py/blob/master/tests/test_payload.py#L87-L111

Comment (majorgreys) [reply tests/test_integration.py:306]: Though I agree we can find a better way to perform such a test.

**Pull request merged**